synapseclient/utils.py

```diff
@@ -154,19 +154,17 @@ def extract_user_name(profile):
 
 
 def _get_from_members_items_or_properties(obj, key):
-    try:
-        if hasattr(obj, key):
-            return getattr(obj, key)
-        if hasattr(obj, 'properties') and key in obj.properties:
-            return obj.properties[key]
-    except (KeyError, TypeError, AttributeError):
-        pass
-    try:
+    if hasattr(obj, key):
+        return getattr(obj, key)
+    if hasattr(obj, 'properties') and key in obj.properties:
+        return obj.properties[key]
+    try :
         if key in obj:
             return obj[key]
         elif 'properties' in obj and key in obj['properties']:
```

Comment:
try:
    if key in obj:
        return obj[key]
    else:
        return obj['properties'][key]
except (KeyError, TypeError):
    # we cannot get the key from this obj, so this case will be treated as key not found
    pass